x/auth/middleware/middleware.go

```diff
@@ -97,13 +97,23 @@ func NewDefaultTxHandler(options TxHandlerOptions) (tx.Handler, error) {
 		TxTimeoutHeightMiddleware,
 		ValidateMemoMiddleware(options.AccountKeeper),
 		ConsumeTxSizeGasMiddleware(options.AccountKeeper),
+		// `DeductFeeMiddleware` and `IncrementSequenceMiddleware` should be put outside of `WithBranchedStore` middleware,
+		// so the storage writes are not discarded when tx fails.
 		DeductFeeMiddleware(options.AccountKeeper, options.BankKeeper, options.FeegrantKeeper),
+		IncrementSequenceMiddleware(options.AccountKeeper),
```

Comment:
IncrementSequenceMiddleware increments the seq. We should:
- either run IncrementSequenceMiddleware after sig verification
- or inside IncrementSequenceMiddleware, run `incrementSeq` after next.DeliverTx

I think this will solve most failing tests

Reply:
thanks, changed, let's see how it runs.

Reply:
hmm OK, but now `SetPubKeyMiddleware` and `ValidateSigCountMiddleware` need to be before `SigVerificationMiddleware`.

Can we actually put WithBranchedStore towards the end, so that at the end we have:
```
// ...all other middlewares except tips
WithBranchedStore
ConsumeBlockGasMiddleware
NewTipMiddleware
```